File: Interfaces/IProductPricingGroupByService.cs
using cms_webapi.DTOs;

namespace cms_webapi.Interfaces
{
    public interface IProductPricingGroupByService
    {
        Task<ApiResponse<List<ProductPricingGroupByDto>>> GetAllProductPricingGroupBysAsync();
        Task<ApiResponse<ProductPricingGroupByDto>> GetProductPricingGroupByByIdAsync(int id);
        Task<ApiResponse<ProductPricingGroupByDto>> CreateProductPricingGroupByAsync(CreateProductPricingGroupByDto createDto);
        Task<ApiResponse<ProductPricingGroupByDto>> UpdateProductPricingGroupByAsync(int id, UpdateProductPricingGroupByDto updateDto);
        Task<ApiResponse<object>> DeleteProductPricingGroupByAsync(int id);
    }
}

File: WebApi/Interfaces/IProductPricingGroupByService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { ProductPricingGroupByDto, CreateProductPricingGroupByDto, UpdateProductPricingGroupByDto } from '../Models/ProductPricingGroupByDto';

export interface IProductPricingGroupByService {
  getAllProductPricingGroupBys(): Promise<ApiResponse<ProductPricingGroupByDto[]>>;
  getProductPricingGroupByById(id: number): Promise<ApiResponse<ProductPricingGroupByDto>>;
  createProductPricingGroupBy(createDto: CreateProductPricingGroupByDto): Promise<ApiResponse<ProductPricingGroupByDto>>;
  updateProductPricingGroupBy(id: number, updateDto: UpdateProductPricingGroupByDto): Promise<ApiResponse<ProductPricingGroupByDto>>;
  deleteProductPricingGroupBy(id: number): Promise<ApiResponse<object>>;
}